bundles/alpha.model/src/alpha/model/transformation/Normalize.xtend

```diff
@@ -409,12 +413,17 @@ class Normalize extends AbstractAlphaCompleteVisitor {
 		
 		val origContainer = be.eContainer  as AlphaCompleteVisitable 
 		
-		binaryExpressionRules(be, be.left)
-		binaryExpressionRules(be, be.right)
+		val bothCase = be.left instanceof CaseExpression && be.right instanceof CaseExpression
+		if (!bothCase) {
+			binaryExpressionRules(be, be.left)
+			binaryExpressionRules(be, be.right)
+		} else {
+			binaryExpressionRules(be, EcoreUtil.copy(be.left))
```

Comment:
Good catch! Yes, I put this in there because it resolves an EMF containment issue that appears to happen sometimes when binary expressions involve two case expressions. I haven’t pinpointed the root cause yet, but this logic avoids the containment issues. I opened issue #44 to track this.